app/utils/blocks.js

```diff
@@ -95,15 +95,25 @@ export function clearValidationsFor(block) {
   block.setWarningText(null)
 }
 
-export function addWarning(block, message, index, itemChar = '☆ ') {
-  block.setWarningText(itemChar + lineWrap(message), index)
+const setWarningBubbleColour = (block, colour) => {
+  const unBoundedSetVisible = Blockly.Warning.prototype.setVisible
+  const boundedSetVisible = unBoundedSetVisible.bind(block.warning)
+  block.warning.setVisible = (visible) => { boundedSetVisible(visible); if (visible) block.warning.bubble_.setColour(colour) }
```

Comment:
El problema surge de que cada vez que se hace click en el ⚠️ se crea/destruye el bubble, no es que se lo oculta. Por esto pasaba que solo la primera vez que aparecia el bubble tenia el color correcto, porque se lo creaba y se le cambiaba el color justo ahi. Al destruirlo y volverlo a crear el bubble tiene el color default porque no se le cambia el color.

Entonces la idea es que cada vez que se clickee el ⚠️ se le cambie el color al bubble recien creado, si es que el click que se hace es para que aparezca el bubble.

Cada vez que se hace click en el ⚠️ de un bloque se ejecuta el metodo `setVisible` del `block.warning`. Esto es algo que hace Blockly, asi que no podemos tocar la logica del click de ⚠️, por lo menos que yo sepa. Blockly no nos da algo como `onWarningClick`, asi que me tuve que colgar de lo que sé que se ejecuta al clickear el ⚠️, osea el `setVisible`.

Entonces lo que hice es redefinir el setVisible para que haga todo lo que venia haciendo antes, y agregarle la linea esa del `setColour`, para que se ejecute cada vez que se haga click, como un super().

El `bind` clona la funcion y le clava como `this` de la funcion el parametro que se le pasa, asi puedo tener en la mano el `setVisible` que se viene usando.


mucho texto perdon

Reply:
Podés usar [`apply`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply) para eso:
```suggestion
  block.warning.setVisible = (visible) => { 
    Blockly.Warning.prototype.setVisible.apply(block.warning, visible)
    if (visible) block.warning.bubble_.setColour(colour) 
  }
```

ATENDEME ese uso de contextos de ejecución por parte de @tfloxolodeiro . ¿Quién necesita guardarse un color cuando puedo guardar una función que lo tiene ya parametrizado?

Reply:
Creo que queremos utilizar `call` en lugar de `apply` por [este detalle](https://stackoverflow.com/questions/1986896/what-is-the-difference-between-call-and-apply).

Reply:
Usando call o apply no me esta funcando, explota al ejecutar 2 veces por:
![image](https://user-images.githubusercontent.com/37090248/178817658-142fd1c7-a464-4c7c-a628-192b1dbc7e76.png)

La primera linea del stack es inentendible, pero la segunda del stack es:
![image](https://user-images.githubusercontent.com/37090248/178818389-f4d4a3de-ca94-4648-b109-cfbda5baf26b.png)


No le dedique mucho tiempo para tratar de entender por que explota de esta forma, pero podria hacerlo 👀

Reply:
Que raro, yo lo probé y anduvo bien (con Firefox, no se si será relevante).